lib/pry/commands/play_motion_picture.rb

```diff
@@ -0,0 +1,234 @@
+    [
+      'https://www.youtube.com/watch?v=4zLfCnGVeL4',
+      'The Sound of Silence (Original Version from 1964)'
+    ],
+    ['https://www.youtube.com/watch?v=3Bv27OcVlMQ', 'Chezidek - Call Pon Dem'],
+    ['https://www.youtube.com/watch?v=co6WMzDOh1o', 'U2 - Beautiful Day'],
+    ['https://www.youtube.com/watch?v=OP-A16Uybaw', 'Michael Collins 1996 - pułapka'],
+    ['https://www.youtube.com/watch?v=1OEAYjo606A', 'Sway - Still Speedin'],
+    ['https://www.youtube.com/watch?v=1IeLb87K-so', 'Aslan - Angie (live in Vicar Steet - Dublin)'],
+    [
+      'https://www.youtube.com/watch?v=e3-5YC_oHjE',
+      'U2 - I Still Haven\'t Found What I\'m Looking For'
+    ],
+    ['https://www.youtube.com/watch?v=iSHy01D_DEg', 'Billy Connolly - Men and Women'],
+    ['https://www.youtube.com/watch?v=gqSzDJGFCgI', 'Thin Lizzy - Bad Reputation'],
+    ['https://www.youtube.com/watch?v=aIXyKmElvv8', 'The Fugees - Ready or Not'],
+    [
+      'https://www.youtube.com/watch?v=YIizdazLTeU',
+      'The Beatles - With a Little Help From My Friends'
+    ],
+    [
+      'https://www.youtube.com/watch?v=k2C5TjS2sh4',
+      'Roxette - It Must Have Been Love'
+    ],
+    [
+      'https://www.youtube.com/watch?v=7ogBRXHGexo',
+      'The Lady who made President Ahmadinejad Cry [HQ]'
+    ],
+    [
+      'https://www.youtube.com/watch?v=iyFijjikkeM',
+      'E.T. Phone Home.flv'
+    ],
+    [
+      'https://www.youtube.com/watch?v=3J8p-NMMI3E',
+      'Billy Connolly Introduces Bob, Jo and Margaret'
+    ]
+  ].freeze
+
+  MOTION_PICTURES_SEQ = MOTION_PICTURES.dup.to_enum
+  private_constant :MOTION_PICTURES, :MOTION_PICTURES_SEQ
+
+  match 'play-motion-picture'
+  group 'Misc'
+  description 'Bored? Play a motion picture.'
+  command_options argument_required: false
+
+  def options(o)
+    o.on :m, :match,
+         'Play first motion picture that matches through a fuzzy search.',
+         argument: true
+    o.on :s, :sequential,
+         'Play motion pictures in sequential order. Rewinds to start at the end.',
+         argument: false
+    o.on :i, :index,
+         'Play a motion picture at a specific index.',
+         argument: true
+    o.on :c, :count,
+         'Print the number of motion pictures available to play.',
+         argument: false
+  end
+
+  def process
+    case
+    when opts.count?
+      process_count MOTION_PICTURES.size
+    when opts.index?
+      process_index opts[:i]
+    when opts.sequential?
+      process_seq MOTION_PICTURES_SEQ
+    when opts.match?
+      process_fuzz_match opts[:m]
+    else # Play motion picture at random.
+      play
+    end
+  end
+
+  private
+  def process_count(count)
+    _pry_.pager.page '%s motion pictures available' % count
+  end
+
+  def process_index(int)
+    raise IndexError, "unsigned integer required (>= 0)" if int.to_i < 0
+    play MOTION_PICTURES[Integer(int)][0]
+  end
+
+  def process_seq(seq)
+    play seq.next[0]
+  rescue Pry::RescuableException
```

Comment:
why not rescue the more specific `StopIteration` exception here rather than a more general one? or could another exception be raised?

Reply:
true, will update for only StopIteration.

Reply:
e442b0b095190fca2e2b020d9946d6dc90a1c86c